text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,363 @@
+
+```typescript
+myServiceNetwork.addLatticeAuthPolicy(
+  [
+    new iam.PolicyStatement({
+      principal: new iam.AccountPrincipal('12345678900')
+    }),
+  ]
+);
+  iam.policyDocument: iam.PolicyStatement[]): void;
+```
+The service can be configured to log to S3, Stream to kinesis, or use Cloudwatch with the `.logToS3()`, `sendToCloudWatch()` or `streamtoKinesis()` methods. 
+
+```typescript
+const loggingbucket: S3.Bucket
+myServiceNetwork.logToS3(loggingbucket)
+```
+
+The service Network can be Shared using RAM using the `.share()` method
+
+```typescript
+myServiceNetwork.share({
+  name: 'myserviceshare',
+  allowExternalPrincipals: false,
+  principals: new iam.AccountPrincipal('12345678900')
+})
+```
+In order to access a service network a vpc must be associated with a vpc. 
+Optionally security groups can be applied. 
+For vpcs, in the same account as the Servicenetwork;
+```typescript
+const vpc: ec2.Vpc;
+const securityGroup: ec2.SecurityGroup
+myServiceNetwork.associateVPC(vpc, [securityGroups])
+})
+```
+Cross account 
+```typescript
+const vpc: ec2.Vpc;
+const securityGroup: ec2.SecurityGroup;
+cosnt serviceNetwork = ServiceNetwork.importFromName('mylatticenetwork')
+myServiceNetwork.associateVPC(vpc, [securityGroups])
+```
+
+#### Create a service from a Lambda and associate it with the Service
+
+A lattice service is the 'front' for various applicaitons that might be 'served' by AWS resources such
+as ec2 Instances, Applicaiton Load Balancers, Lambdas.   Listeners are attached to the service, which have rules and targets. 
+
+```typescript
+import { Service, Protocol, FixedResponse } from '@aws-cdk/aws-vpclattice-alpha';
+
+
+const myService: Service = new Service(this, 'myservice', {
+  name: 'myservice',
+})
+```
+
+Add a listner to the service, with a certificate, a hostname and DNS
+```typescript
+const myCertificate: certificate_manager.Certificate
+const serviceListener: lattice.Listener = myService.addListener(
+  defaultAction: FixedResponse.NOT_FOUND
+  protocol: Protocol.HTTPS,
+  name: 'MyServiceListener'
+)
+myService.addCertificate(myCertificate);
+myService.addCustomDomain('example.org');
+myService.addDNSEntry('abcdef');
+```
+
+Create targets which will provide the 'content' for your service, in this example
+we target a lambda. Then use the target in a Rule that is attached to the Listener.
+
+```typescript
+const functionOne: aws_lambda.Function
+
+const targetOne = new lattice.LatticeTargetGroup(this, 'TargetOne', {
+  name: 'targetgroupOne',
+  lambdaTargets: [
+    functionOne
+  ],
+})
+
+serviceListener.addListenerRule({
+  name: 'listentolambdaone',
+  action: [{ target: targetOne }],
+  priority: 100,  
+  pathMatch:  {
+    pathMatchType: lattice.PathMatchType.EXACT,
+    matchValue: '/serviceOne',
+    caseSensitive: false,
+  } 
+})
+    
+```
+3. Finally, add the Service to the the Service Network
+```typescript
+serviceNetwork.addService(myService)
+```
+
+## API Design
```

Comment:
I'm not seeing what the proposed props are for any of the Constructs below so it's hard to understand what the user experience will be here. I like the convenience functions, but what is the contract if a user wants to define the various sets of functionality as props when they declare the construct initially?